### PR TITLE
PdoAdapter: cannot reconnect in the middle of transaction

### PR DESCRIPTION
**Pull request by bryanlatten**
This would cause a transaction to be partially committed --- not good.

### CODE REVIEW

**Comment by mikesherov** [comment]
Consider `isInTransaction` (and same for member variable) to better communicate Boolean.

Otherwise...

**Comment by mikesherov** [comment]
:+1:

**Comment by bryanlatten** [comment]
@mikesherov fixed function naming, initially thought it looked ugly -- but, that is the pattern

**Comment by ingluisjimenez** [comment]
:+1:

File: tests/unit/Adapters/PdoAdapterTest.php [hunk 1]
```diff
@@ -184,36 +183,49 @@ public function queryReconnect( $master ) {
...
-  public function queryReconnectBad( $master ) {
+  public function queryReconnectBad( $master, $in_transaction ) {
 
     $sql        = "SELECT * FROM abc WHERE def = ? && ghi = ?";
     $params     = [ 123, 456 ];
     $connect_fx = ( $master )
                   ? '_getMasterAdapter'
                   : '_getReplicaAdapter';
-
+    $methods    = [ 'isInTransaction', '_getMasterAdapter', '_getReplicaAdapter', '_reconnectAdapter' ];
```
Comment (jimdelois) [inline tests/unit/Adapters/PdoAdapterTest.php]: `$methods    = [ 'isInTransaction', '_reconnectAdapter', $connect_fx ];`

Also, remove the second parameter `[ $connect_fx ]` on line `197` as it's confusing and inaccurate.  The `ConnectionService` doesn't have either of `_getMasterAdapter` or _getReplicaAdapter`.

**Comment by jimdelois** [comment]
Consider augmenting the `beginTransaction` test to test that the new flag is set...

```
  /**
   * @test
   */
  public function beginTransaction() {

    $adapter = $this->_getDisabledMock( PdoAdapter::class, [ '_getMasterAdapter' ] );
    $pdo     = $this->_getDisabledMock( PDO::class, [ 'beginTransaction' ] );

    $adapter->expects( $this->once() )
      ->method( '_getMasterAdapter' )
      ->will( $this->returnValue( $pdo ) );

    $pdo->expects( $this->once() )
      ->method( 'beginTransaction' )
      ->will( $this->returnValue( true ) );

    $this->assertTrue( $adapter->beginTransaction() );

    // This is the only NEW line to add
    $this->assertTrue( $adapter->isInTransaction() );

  } // beginTransaction
```

In a similar fashion, add `$this->assertFalse( $adapter->isInTransaction() );` to the test `rollBack` and `commit`.

You're mocking the SUT so nothing else is asserting that these are actually working correctly.  Updating these tests will fill that gap.

**Comment by jimdelois** [comment]
:+1:

**Pull request merged**